### PR TITLE
Add Employee Scaffolds

### PR DESCRIPTION
**Pull request by jestradadeveloper**
# Description
This commit add scaffold for Employee Entity, this action is responsible for:
 - Creation of route for Employee resources.
 - Creation of views for Employee CRUD
 - Creation of model Employee with validations
 - Creation of EmployeeController with CRUD actions and authentication and authorization restrictions
 - Testing validations for model, views, controller , routing and request .

 Examples:
 - What?:  CRUD of Employee implementation 
 - Why?: To allow to the user admin add, update and edit Employees for the app

 # Testing
Running rspec command should validate all of possible scenarios for this Model.

  ## Screenshots
Captura de Pantalla 2021-10-14 a la(s) 10 33 28
Captura de Pantalla 2021-10-14 a la(s) 10 34 04
Captura de Pantalla 2021-10-14 a la(s) 10 34 22

### CODE REVIEW

File: app/views/employees/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  <div class="field">
+    <%= form.label :private_number %>
+    <%= form.number_field :private_number %>
+  </div>
+
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
```
Comment (YilKanda) [inline app/views/employees/_form.html.erb:32]: I recommend that you style the forms.

File: app/views/employees/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        <td><%= employee.private_number %></td>
+        <td><%= link_to 'Show', employee %></td>
+        <td><%= link_to 'Edit', edit_employee_path(employee) %></td>
+        <td>
+          <%= button_to 'Delete', employee_path(employee), :method => 'delete', :confirm => 'Are you sure?' %>
+       </td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
```
Comment (YilKanda) [inline app/views/employees/index.html.erb:29]: I recommend that you style this part of the code where all the employees are shown.

File: app/views/employees/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  <%= @employee.private_number %>
+</p>
+
+<p>
+  <strong>User:</strong>
+  <%= @employee.user_id %>
+</p>
+
+<%= link_to 'Edit', edit_employee_path(@employee) %> |
+<%= link_to 'Back', employees_path %>
```
Comment (YilKanda) [inline app/views/employees/show.html.erb:24]: I recommend that you style this part of code where the information of a single employee is displayed.

File: app/views/employees/_form.html.erb (hunk 1 above)
Comment (jestradadeveloper) [reply app/views/employees/_form.html.erb:32]: The styles for all this scaffold elements are going to be in a different issued.

File: app/views/employees/index.html.erb (hunk 2 above)
Comment (jestradadeveloper) [reply app/views/employees/index.html.erb:29]: @YilKanda  The styles for all this scaffold elements are going to be in a different issued.

File: app/views/employees/show.html.erb (hunk 3 above)
Comment (jestradadeveloper) [reply app/views/employees/show.html.erb:24]: @YilKanda  The styles for all this scaffold elements are going to be in a different issued.

**Review by JuanequeX — APPROVED** [review]
LGTM

**Review by YilKanda — APPROVED** [review]
LGTM!

**Pull request merged**